common/types/hashable.go

```diff
@@ -0,0 +1,10 @@
+package types
+
+// A chain-agnostic generic interface to represent the following native types on various chains:
+// PublicKey, Address, Account, BlockHash, TxHash
+type Hashable interface {
+	ToBytes() []byte
+	ToString() string
```

Comment:
Interesting that we rarely use these methods in txmgr (I notice we don't need to refactor to use any of these methods in this PR or even call native[Address|Hash]() ). Probably all these methods would be used in the EvmClient?

Reply:
EvmClient will need to convert back into native types.
Orm will need to serialize/deserialize.
Some places in Txmgr will need to compare to values.

Reply:
We don't have to limit ourselves to one monolithic interface for all cases. If we can use simpler, minimal interfaces or funcs in some places then we should. It decouples things and prevents the generics from spreading so much.